### PR TITLE
Add schemathesis, update OpenAPI, update Deutschland-Workflow

### PR DESCRIPTION
**Pull request by t-huyeng**
- Added Schemathesis
  -  with `--hypothesis-phases=explicit`. So that only examples, specified explicitly in the API schema are used. 
- Update OpenAPI
- Update Deutschland-Workflow to only commit and upload code on bundesAPI-git.
- Update Spectral version to `latest`

### CODE REVIEW

File: .github/workflows/deutschland_generator.yaml [hunk 1]
```diff
@@ -31,4 +32,14 @@ jobs:
           upload-to-testpypi: false
           pypi-token: ${{ secrets.PYPI_PRODUCTION }}
           testpypi-token: ${{ secrets.PYPI_TEST }}
-          python-version: ${{ matrix.python-version }}
\ No newline at end of file
+          python-version: ${{ matrix.python-version }}
+
+      - name: "Generate deutschland code"
+        if: github.repository_owner !== 'bundesAPI'
```
Comment (wirthual) [inline .github/workflows/deutschland_generator.yaml]: Can we shorten this to:
```
commit-to-git:  ${{ github.repository_owner == 'bundesAPI'}}
upload-to-pypi: ${{ github.repository_owner == 'bundesAPI'}}
upload-to-testpypi: false
```
within one block?

**Review by wirthual — COMMENTED** [review]
Small improvement to shorten file. We should also add this to the template.

File: .github/workflows/deutschland_generator.yaml (hunk 1 above)
Comment (t-huyeng) [reply .github/workflows/deutschland_generator.yaml]: Yes, looks like in one block is working.  I would prefer this instead an disabling the workflow completely. 

Sidenote: I had the syntax wrong on the initial commits -> `!==` needs to be `!=` in Actions. But since we now only use the short version now, we do not need this anymore.
